### ISSUE TITLE
Phrases in “derived terms” are broken

### ISSUE THREAD

**Issue by GrimPixel** [tatuylonen/wiktextract#573]
One example: [beißen](https://en.wiktionary.org/wiki/bei%C3%9Fen), where the phrase “Hunde, die bellen, beißen nicht” is split into three dicts:
```
{'word': 'Hunde', '_dis1': '0 0 0'}, {'word': 'die bellen', '_dis1': '0 0 0'}, {'word': 'beißen nicht', '_dis1': '0 0 0'}
```

**Comment by xxyzz** [comment]
I guess this is because the code at here https://github.com/tatuylonen/wiktextract/blob/6f54f71ebcf9497d1e15433cd11e58fdfa16c2a2/src/wiktextract/linkages.py#L212 processes the list item node's plain text but doesn't use the wikitext node structure?

**Comment by kristian-clausal** [comment]
Doing parsing with some understanding of node structure would be amazing, but it's extremely tricky.

Also, if there is no node structure (that is, a link) to rely on, we would still have to rely on the text parsing.

We can't do text parsing using nodes, because we are relying on Python's low-level string and regex functions to keep the project... not performant, but non-geological in time expenditure. `re` doesn't care about nodes, and neither do any other string methods and functions. We'd probably have to write C libraries just for wikitextprocessor/wiktextract to have some kind of multilayered node-aware string object type... It could be done with Python classes in Python, but it would be slow.

We could try something like checking all links ahead of time and create a list of links for each page, then using that data in cases like this... But it would be costly. I guess we could cache link data when parsing the page. But another issue is also that you can't distinguish 100% whether a word link is an actual word link or a link to a linguistic term, because those use the same internal links. We can do it heuristically, but sometimes you have words that have genuine word-links to linguistic terms because the article itself is related, &soforth, &soforth.

**Comment by xxyzz** [comment]
At least for this case, the wikitext is `* {{l|de|Hunde, die bellen, beißen nicht}}` and expands to a single link so the code could "know" these 3 words are not separated. If this is the common layout used in en edition's "Derived terms" section then I guess we could use this node structure information? For example, we could try to find the "l" template in the list and expand the template or use the parameter, then fallback to parse list's text if the page doesn't use this layout.

**Comment by kristian-clausal** [comment]
I made a PR with a fix, but it breaks other tests.

**Comment by xxyzz** [comment]
Fixed by #584
